tests/Doctrine/Tests/Mocks/ConnectionCommitFailMock.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Doctrine\Tests\Mocks;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Statement;
+
+/**
+ * Mock class for Connection.
+ */
+class ConnectionCommitFailMock extends ConnectionMock
```

Comment:
Not sure why nobody commented here yet, but this was talked about a bit internally and consensus seemed to be that doctrine team doesn't like dedicated mock class for such simple, non-reusable mock like is a case here. PHPunit mock was recommended to be used instead, like
```php
$this->_connectionMock = $this->getMockBuilder(ConnectionMock::class)
  ->setConstructorArgs([[], new DriverMock()])
  ->setMethods(['commit'])
  ->getMock();

$this->_connectionMock->method('commit')->willReturn(false);
```

although if you prefer, I would also suggest anonymous class such as
```php
$this->_connectionMock = new class([], new DriverMock()) extends ConnectionMock {
  public function commit()
    {
        return false;
    }
};
```
I think complaint was mainly about polluting project/namespace with one-off classes.

Reply:
I wouldn't use an anonymous class extending one of those Mock classes because this will still have the disadvantages mentioned in doctrine/dbal#3470. I would prefer a cut and that we break with the usage of custom Mock classes, so that it will be easier to remove them in another PR.

Reply:
Only first point from https://github.com/doctrine/dbal/pull/3470 apply here. And refactoring of signatures via phpstorm is piece of cake, unlike handling stuff like method renames and change of constructors in phpunit mocks. And readability, SA analysis support. But since we already know type hints are going to be added, it indeed kinda makes sense to use phpunit mock here to avoid having to touch this in upper branches.

Reply:
The third point also apply here for implicit dependencies between tests using `ConnectionMock`. Invocations won't be tracked on it either. I see more advantages between those two solutions in the phpunit mock.

Reply:
that point applies for external classes, not for anonymous ones used in single test case

Reply:
Ok for me